### PR TITLE
WIP: Display memory usage in DevTools.

### PR DESCRIPTION
**Pull request by grokys**
## What does the pull request do?

In an attempt to get to the bottom of #3983 I've added some memory diagnostics to DevTools:

image

- Total memory: The total memory of the application ([Process.PrivateMemorySize64](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process.privatememorysize64?f1url=%3FappId%3DDev16IDEF1%26l%3DEN-US%26k%3Dk(System.Diagnostics.Process.PrivateMemorySize64);k(DevLang-csharp)%26rd%3Dtrue&view=netcore-3.1))
- Managed memory: The managed memory ([GC.GetTotalMemory(true)](https://docs.microsoft.com/en-us/dotnet/api/system.gc.gettotalmemory?f1url=%3FappId%3DDev16IDEF1%26l%3DEN-US%26k%3Dk(System.GC.GetTotalMemory);k(DevLang-csharp)%26rd%3Dtrue&view=netcore-3.1))
- Graphics memory: The Skia graphics memory ([GrContext.dumpMemoryStatistics](https://api.skia.org/classGrContext.html#a3543f1cf044b39fbbfae96cf92e2cb22))
- Dark matter: Unexplained memory usage (total memory minus managed and graphics memory)

### CODE REVIEW

**Comment by pr8x** [comment]
Nice, in the future it would be cool to get additional info like like WPF Perforator does:

https://docs.microsoft.com/en-us/previous-versions/aa969767(v=vs.110)?redirectedfrom=MSDN#perforator-graphs

https://docs.microsoft.com/en-us/previous-versions/aa969767(v=vs.110)?redirectedfrom=MSDN#visual-element-tree

Showing inclusive/exclusive layout time would be awesome as well :)

File: src/Avalonia.Visuals/Platform/IGraphicsMemoryDiagnostics.cs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+﻿namespace Avalonia.Platform
+{
+    public interface IGraphicsMemoryDiagnostics
```
Comment (pr8x) [inline src/Avalonia.Visuals/Platform/IGraphicsMemoryDiagnostics.cs:3]: Public interface should have documentation :)

**Comment by pr8x** [comment]
You're using `axaml` already, I am wondering if we shouldn't wait with that and change all xaml files to be axaml eventually?

**Comment by pr8x** [comment]
`PerformanceViewModel` should probably be called `PerformancePageViewModel`

**Comment by grokys** [comment]
Going to close this for the moment as this info just isn't turning out to be that useful in the end.

**Pull request closed without merging**